src/portfft/descriptor.hpp

```diff
@@ -491,18 +509,94 @@ class committed_descriptor {
     // Dummy parameter is needed as only partial specializations are allowed without specializing the containing class
     template <detail::level Lev, typename Dummy>
     struct inner {
-      static Scalar* execute(committed_descriptor& desc, dimension_struct& dimension_data);
+      static Scalar* execute(committed_descriptor& desc, std::vector<kernel_data_struct>& kernels);
     };
   };
 
   /**
    * Calculates twiddle factors for the implementation in use.
    *
-   * @param dimension_data data about the dimension for which twiddles are needed
+   * @param kernels vector of kernels
    * @return Scalar* USM pointer to the twiddle factors
    */
-  Scalar* calculate_twiddles(dimension_struct& dimension_data) {
-    return dispatch<calculate_twiddles_struct>(dimension_data.level, dimension_data);
+  Scalar* calculate_twiddles(detail::level level, std::vector<kernel_data_struct>& kernels) {
+    return dispatch<calculate_twiddles_struct>(level, kernels);
+  }
+
+  /**
+   * Sets the specialization constants for all the kernel_ids contained in the vector
+   * returned from prepare_implementation
+   * @tparam SubgroupSize Subgroup size
+   * @param top_level selected level of implementation
+   * @param prepared_vec vector as returned from prepare_implementation
+   * @param compute_direction direction of compute, as in forward or backward
+   * @param dimension_num which dimension are the kernels being built for
+   * @param is_compatible flag to be set if the kernels are compatible
```

Comment:
Is this an out parameter? If so  it should be documented as such. If not there is no need for it to be a reference.

Compatible with what? Extend the documentation.

Reply:
Removed is_compatible altogether

Reply:
You may have not pushed that changed yet

Reply:
update the doxygen to match

Reply:
Done